packages/cadl-ranch-specs/cadl-ranch-summary.md

```diff
@@ -221,6 +221,27 @@ Expect error code 403 and error body:
 }
 ```
 
+### Authentication_Http_valid
```

Comment:
This should probably be Authentication.Http.sharedAccessKey (or without http)

Reply:
Technically, I used "SharedAccessKey" since this is the actual value that EventGrid needs, but the test is designed to just confirm that the value passed as "scheme" is handled correctly. IOW, if I were to write "CogSevKey", this should work as well. It's why I tried to avid putting SharedAccessKey into the name.

Maybe Authentication.Http.CustomScheme ?

Reply:
What do we expect the client to generate 

new SharedAcessKeyAuth("…")

or 

new HttpAuth("SharedAccessKey", "…")

for the first one I think we should have a test dedicated to it. For the 2Ms I guess it’a ok

Reply:
I expect to generate something like:
client = client(endpoint, AzureKeyCredential("mykey"))

The fact that the scheme is "SharedAccessKey" is a hidden detail

Reply:
I guess the question then is lets assume we have this other http scheme called `MasterKey`, would the Client generated code automatically inject that.

Basically do we expect this:
```
# In generted client.py
.headers["Authentication] = httpScheme + " " + value;
# where httpScheme would be whatever is the value in the spec
```
and the user would use that
```
client = client(endpoint, AzureKeyCredential("mykey"))
```


What do you expect those 2 things too look like if the scheme was `Basic`, where the value should be `username:password`

Reply:
If it was `Basic`, I would not generate that, I'll definitely take a different credentials. The same way when it's "Bearer", we take a TokenCredential.

I guess it means there are some special values mapped to special types. I guess what I'm doing here, is to say that the fallback for unknown scheme is AzureKeyCredential.

Reply:
I see, then yeah having this custom scheme naming then should be good. That would assume we can have this structure right:
```
Authentication
  Http:
     Bearer
     Basic
     Custom
  OAuth2
  ApiKey
    +